src/main/java/com/thealgorithms/sorts/CountingSort.java

```diff
@@ -0,0 +1,54 @@
+package com.thealgorithms.sorts;
+
+import java.util.Arrays;
+
+/**
+ * A standard implementation of the Counting Sort algorithm for integer arrays.
+ * This implementation has a time complexity of O(n + k), where n is the number
+ * of elements in the input array and k is the range of the input.
+ * It works only with integer arrays.
+ *
+ * The space complexity is O(k), where k is the range of the input integers.
+ *
+ * Note: This implementation does not handle negative integers as it
+ * calculates the range based on the minimum and maximum values of the array.
+ *
+ */
+public final class CountingSort {
+    private CountingSort() {
+    }
+
+    /**
+     * Sorts an array of integers using the Counting Sort algorithm.
+     *
+     * @param array the array to be sorted
+     * @return the sorted array
+     */
+    public static int[] sort(int[] array) {
+        if (array.length == 0) {
+            return array;
+        }
+
+        final int max = Arrays.stream(array).max().orElse(Integer.MIN_VALUE);
+        final int min = Arrays.stream(array).min().orElse(Integer.MAX_VALUE);
```

Comment:
At this point `array` cannot be empty, therefore, I would suggest to use `getAsInt()`.